common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -419,7 +419,8 @@ protected void wakeup(boolean inEventLoop) {
 
     @Override
     public boolean inEventLoop(Thread thread) {
-        return thread == this.thread;
+        return thread == this.thread ||
+                executor instanceof EventExecutor && ((EventExecutor) executor).inEventLoop(thread);
```

Comment:
@louiscryan - Just an update on this...I'm going to consult with @normanmaurer to get his thoughts.